### PR TITLE
[air] add bulk ingest benchmarks

### PR DESCRIPTION
**Pull request by matthewdeng**
Signed-off-by: Matthew Deng <matt@anyscale.com>

## Why are these changes needed?

Adding benchmark release test for Bulk Ingest and benchmark results.

Successful run: https://buildkite.com/ray-project/release-tests-pr/builds/9630#01820536-0c2a-4f52-b960-a7a5cd3ef0af

Benchmark documentation: https://ray--26618.org.readthedocs.build/en/26618/ray-air/benchmarks.html

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/air_tests/air_benchmarks/workloads/data_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        datasets={"train": dataset},
+        preprocessor=dummy_prep,
+        num_epochs=1,
+        prefetch_blocks=1,
+        dataset_config={"train": DatasetConfig(split=True)},
+    )
+    trainer.fit()
+
+
+# TODO: Verify this runs at scale.
```
Comment (ericl) [inline release/air_tests/air_benchmarks/workloads/data_benchmark.py:53]: Let's remove the streaming code for now?

File: release/air_tests/air_benchmarks/workloads/data_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    record_dim = 1280
+    record_size = record_dim * 8
+    num_records = int(total_size / record_size)
+    dataset = ray.data.range_tensor(num_records, shape=(record_dim,))
+    print("Created dataset", dataset, "of size", dataset.size_bytes())
+    return dataset
+
+
+# TODO: Verify this runs at scale.
+def run_ingest_bulk(dataset, num_workers):
```
Comment (ericl) [inline release/air_tests/air_benchmarks/workloads/data_benchmark.py:39]: Remove TODOs.

File: release/air_tests/air_benchmarks/workloads/data_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        preprocessor=dummy_prep,
+        num_epochs=1,
+        prefetch_blocks=1,
+        dataset_config={"train": DatasetConfig(split=True, use_stream_api=True, stream_window_size=num_workers * GiB)},
+    )
+    trainer.fit()
+
+
+# TODO: Verify this runs at scale.
+def run_infer_bulk(dataset, num_workers, post=None):
```
Comment (ericl) [inline release/air_tests/air_benchmarks/workloads/data_benchmark.py:69]: Let's also remove the inference code for now.

**Comment by ericl** [comment]
Let's keep this file focused only on bulk ingest for now.

**Comment by matthewdeng** [comment]
@ericl ack sounds good in that case I'll update it all in one go with the running test!

File: doc/source/ray-air/benchmarks.rst [hunk 4]
```diff
@@ -3,6 +3,57 @@ AIR Benchmarks
...
+-----------
+
+This task uses the DummyTrainer module to ingest 200GiB of synthetic data.
+
+We test out the performance across different cluster sizes.
+
+- `Bulk Ingest Script`_
+- `Bulk Ingest Cluster Configuration`_
+
+For this benchmark, we configured the nodes to have reasonable disk size and throughput to account for object spilling.
```
Comment (richardliaw) [inline doc/source/ray-air/benchmarks.rst:16]: maybe say what the # workers are for (they're for consumption right?)

**Review by richardliaw — APPROVED** [review]

File: doc/source/ray-air/benchmarks.rst (hunk 4 above)
Comment (ericl) [reply doc/source/ray-air/benchmarks.rst:16]: Nice. Should we add some known limitations section for some of these benchmarks? Such as: it gets much slower if data doesn't fit in memory.

**Pull request merged**